chia/server/start_service.py

```diff
@@ -271,20 +273,7 @@ async def wait_closed(self) -> None:
         self._log.info(f"Service {self._service_name} at port {self._advertised_port} fully closed")
 
 
-async def async_run_service(*args, **kwargs) -> None:
-    # This will get relocated to the source of the kwargs definitions and the original
-    # configuration loading which will make it much more reasonable.
-    initialize_logging(
-        service_name=kwargs["service_name"],
-        logging_config=load_config(kwargs["root_path"], "config.yaml", kwargs["service_name"])["logging"],
-        root_path=kwargs["root_path"],
-    )
-    service = Service(*args, **kwargs)
-    await service.setup_process_global_state()
-    return await service.run()
-
-
-def run_service(*args, **kwargs) -> None:
+def async_run(coro: Coroutine[object, object, T]) -> T:
```

Comment:
```suggestion
def async_run(coro: Awaitable[int]) -> int:
```

Why not just ^? i.e. why is the `Coroutine` hinting required here and why the `TypeVar`?

Reply:
So that would be ok now, but only because typeshed has fixed the hints since the last released version in mypy, or something.  But they have fixed it and `Awaitable` does not satisfy `Coroutine`.  `asyncio.run()` is about processing a coroutine, not awaiting an awaitable.  I think...

https://github.com/python/typeshed/commit/eab82c838a395ef0323d930b8047d037e18832a5#diff-816a4437e484a5998b1afef4f0e0e18aca85e6399b877f3fd883e30e0b4e2010

Reply:
Okay i see, and the `TypeVar` is there to make it some generic helper? Just wondered because with the service start fun we will probably not return anything else than `int` right?